core/src/test/java/org/springframework/security/scheduling/CurrentDelegatingSecurityContextTaskSchedulerTests.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.scheduling;
+
+import org.junit.Before;
+
+import org.springframework.security.core.context.SecurityContext;
+
+/**
+ * Tests using the current {@link SecurityContext} on
+ * {@link DelegatingSecurityContextTaskScheduler}
+ *
+ * @author Giacomo Baso
+ * @since 5.5
+ *
+ */
+public class CurrentDelegatingSecurityContextTaskSchedulerTests
```

Comment:
Is there a reason that these task scheduler test classes should be separated like this? Most Spring Security test classes follow the pattern `XXXTests` where `XXX` is the class under test.

Reply:
Other related test classes follow the same pattern, in particular `AbstractDelegatingSecurityContextExecutorTests` and subclasses. This allows repeating the tests with a different configuration, i.e. to use the current `SecurityContext` or a provided one.

Would you suggest a different approach?

Reply:
Given that those test patterns do not actually check for the use of a specific `SecurityContext` and your comment regarding the mocking of `DelegatingSecurityContextCallable#create`, I've decided to revert to the old tests suite `DelegatingSecurityContextTaskSchedulerTests`, with the addition of a mocked `SecurityContext`.